tuf/signed/verify.go

```diff
@@ -145,3 +138,18 @@ func VerifySignatures(s *data.Signed, roleData data.BaseRole) error {
 
 	return nil
 }
+
+// VerifySignature checks a single signature and public key against a payload
+func VerifySignature(msg []byte, sig data.Signature, pk data.PublicKey) error {
+	// method lookup is consistent due to Unmarshal JSON doing lower case for us.
+	method := sig.Method
+	verifier, ok := Verifiers[method]
+	if !ok {
+		return fmt.Errorf("continuing b/c signing method is not supported: %s\n", sig.Method)
+	}
+
+	if err := verifier.Verify(pk, sig.Signature, msg); err != nil {
+		return fmt.Errorf("continuing b/c signature was invalid\n")
```

Comment:
Non-blocking nitpick:  Maybe this could just be "signature was invalid"?  `VerifySignature` doesn't know it's being called in a loop or that the caller will ignore the error.